test/server/hot_restart_impl_test.cc

```diff
@@ -100,6 +123,98 @@ TEST_F(HotRestartImplTest, DomainSocketError) {
   EXPECT_THROW(std::make_unique<HotRestartImpl>(0, 0, "@envoy_domain_socket", 0), EnvoyException);
 }
 
-} // namespace
+class HotRestartUdpForwardingContextTest : public HotRestartImplTest {
+public:
+  void SetUp() override { setup(); }
+
+  void childRegisterUdpForwardingListener(const Network::Address::Instance& address,
+                                          Network::UdpListenerConfig& listener_config) {
+    hot_restart_->as_child_.registerUdpForwardingListener(address, listener_config);
+  }
+  absl::optional<HotRestartingChild::UdpForwardingContext::ForwardEntry>
+  childGetListenerForDestination(const Network::Address::Instance& address) {
+    return hot_restart_->as_child_.udp_forwarding_context_.getListenerForDestination(address);
```

Comment:
Can we add an interface to access udp_forwarding_context_.getListenerForDestination() in HotRestartingChild instead of declaring friend class?

Reply:
That would be tainting the actual interface in support of tests - if it had that interface it should be a private interface since it's not intended for calling from outside, so testing it would still require the friend class.

Reply:
Would such interface needed in your following changes?
If not, please use peer class instead.

Reply:
Using a peer class now, though I called it a `TestHelper`, does it need to be named `Peer`? (I thought `TestHelper` is more helpfully explicit that it's a test-only relationship.)

Reply:
TestHelper is fine